### PR TITLE
Feat: new custom role resource

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Part of the effort to add custom roles #498.
This PR:
1. updated the schemas.
2. implemented assignment of custom roles for user-project.
3. updated the documentation and examples.
4. added and updated acceptance tests.
5. added additional integration tests.

### CODE REVIEW

File: client/user_project_assignment.go [hunk 1]
```diff
@@ -1,18 +1,18 @@
 package client
 
 type AssignUserToProjectPayload struct {
-	UserId string      `json:"userId"`
-	Role   ProjectRole `json:"role"`
+	UserId string `json:"userId"`
+	Role   string `json:"role" tfschema:"-"`
```
Comment (TomerHeber) [inline client/user_project_assignment.go:5]: changed to string and is now handled manually by setting `tfschema:"-"`.

File: client/user_project_assignment.go [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 }
 
 type UpdateUserProjectAssignmentPayload struct {
-	Role ProjectRole `json:"role"`
+	Role string `json:"role" tfschema:"-"`
```
Comment (TomerHeber) [inline client/user_project_assignment.go:9]: same here.

File: client/user_project_assignment.go [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
 }
 
 type UserProjectAssignment struct {
-	UserId string      `json:"userId"`
-	Role   ProjectRole `json:"role"`
-	Id     string      `json:"id"`
+	UserId string `json:"userId"`
+	Role   string `json:"role" tfschema:"-"`
```
Comment (TomerHeber) [inline client/user_project_assignment.go:14]: same here.

File: env0/resource_team_project_assignment.go [hunk 4]
```diff
@@ -31,9 +31,16 @@ func resourceTeamProjectAssignment() *schema.Resource {
 			},
 			"role": {
 				Type:             schema.TypeString,
-				Description:      "the assigned role (Admin, Planner, Viewer, Deployer)",
-				Required:         true,
+				Description:      "the assigned built-in role (Admin, Planner, Viewer, Deployer)",
+				Optional:         true,
```
Comment (TomerHeber) [inline env0/resource_team_project_assignment.go:35]: now optional with ExactlyOneOf.

File: env0/resource_user_project_assignment.go [hunk 5]
```diff
@@ -75,6 +88,13 @@ func resourceUserProjectAssignmentRead(ctx context.Context, d *schema.ResourceDa
 			if err := writeResourceData(&assignment, d); err != nil {
 				return diag.Errorf("schema resource data serialization failed: %v", err)
 			}
+
+			if client.IsBuiltinProjectRole(assignment.Role) {
```
Comment (TomerHeber) [inline env0/resource_user_project_assignment.go:92]: same variable in the response. However, in the schema, it goes to a different field.

File: env0/resource_user_project_assignment_test.go [hunk 6]
```diff
@@ -91,6 +116,51 @@ func TestUnitUserProjectAssignmentResource(t *testing.T) {
...
+						resource.TestCheckNoResourceAttr(accessor, "role"),
+					),
+				},
+			},
+		}
+
+		runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {
+			gomock.InOrder(
+				mock.EXPECT().AssignUserToProject(projectId, &createCustomPayload).Times(1).Return(&createCustomResponse, nil),
+				mock.EXPECT().UserProjectAssignments(projectId).Times(2).Return([]client.UserProjectAssignment{otherResponse, createCustomResponse}, nil),
```
Comment (Yossi-kerner) [inline env0/resource_user_project_assignment_test.go:156]: Why should it be called twice?

**Review by Yossi-kerner — CHANGES_REQUESTED** [review]
Looks great just left some little comments .We should also test updating a custom role to a built in role and vise versa just to make sure.
Also it seems like you wrote team assignment example in the project assignment file and vise versa.

File: env0/resource_user_project_assignment_test.go (hunk 6 above)
Comment (TomerHeber) [reply env0/resource_user_project_assignment_test.go:156]: There is a "READ" after apply and there is a "READ" before update. (It's how terraform works).

**Comment by TomerHeber** [comment]
> Looks great just left some little comments .We should also test updating a custom role to a built in role and vise versa just to make sure. Also it seems like you wrote team assignment example in the project assignment file and vise versa.

Thanks. I'll fix the examples (Looks like I got them mixed up), and also add additional tests.

**Review by Yossi-kerner — APPROVED** [review]
LGTM

**Pull request merged**